### PR TITLE
10x faster Geocoding, 60% less memory

### PR DESCRIPTION
**Pull request by velvia**
- Use `PackedCoordinateSequence` to efficiently store coordinates, up to 60% less memory usage.  Soft reference for on-demand unpacking to faster form.
- Use `PreparedGeometry` for 10x speedup in actual geo-region-coding operation

### CODE REVIEW

File: src/main/scala/com/socrata/geospace/regioncache/SpatialIndex.scala [hunk 1]
```diff
@@ -18,6 +21,7 @@ import SpatialIndex.Entry
  * @param items A sequence of SpatialIndex.Entry's to index
  */
 class SpatialIndex[T](items: Seq[Entry[T]]) extends Logging {
+  logger.info(s"Created new SpatialIndex with ${items.length} items")
```
Comment (znep) [inline src/main/scala/com/socrata/geospace/regioncache/SpatialIndex.scala]: Is this "created" or "creating"?  It runs when we start creating it, right?

**Comment by znep** [comment]
LGTM, other than minor nit about created/creating.  Also note Codacy's complaint about the `.get`s you added in the tests.  I don't really care about them, but if we want to use Codacy we should either have it setup to not complain on them in tests or we should not use get in tests this way.

File: README.md [hunk 2]
```diff
@@ -84,7 +84,13 @@ pressure, it will attempt to start uncaching regions from the biggest currently
 cached one in order down to try to free memory.  If it doesn't work then it will
 throw an error.
 
-One good region to test memory usage with is nationwide zip codes: https://data.austintexas.gov/d/a3it-2a2z with ~34000 features.
+One good region to test memory usage with is nationwide zip codes: https://data.austintexas.gov/d/a3it-2a2z with ~34000 features.  Another one is: qz3q-ghft.
```
Comment (urmilan) [inline README.md]: Can you specify if qz3q-ghft is a production dataset or in some other environment?

**Comment by velvia** [comment]
@znep @urmilan thanks, fixed stuff, including the Codacy issue by using ScalaTest's `OptionValues` trait.

**Pull request merged**